File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    //values
    int wordCount = 1;
    int letterCount = 0;
    int sentenceCount = 0;
    string userIn = get_string("Text: ");
    int len = strlen(userIn);

    //analyze the text
    for (int i = 0; i < len; i++)
    {
        if (userIn[i] == ' ')
        {
            wordCount++;
        }
        else if (userIn[i] == '!' || userIn[i] == '?' || userIn[i] == '.')
        {
            sentenceCount++;
        }
        else if (isalpha(userIn[i]) != 0)
        {
            letterCount++;
        }


    }

    float letterAvg = ((float)letterCount / (float)wordCount) * 100;
    float sentenceAvg = ((float)sentenceCount / (float)wordCount) * 100;

    int grade = round(0.0588 * letterAvg - 0.296 * sentenceAvg - 15.8);

    //special formatting
    if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

def main():
	text = get_string("Text: ")

	sentence_enders = [".", "!", "?"]
	letter_count = 0
	sentence_count = 0
	word_count = 1

	for char in text:
		if char.isalpha():
			letter_count += 1

		elif char == " ":
			word_count += 1

		elif char in sentence_enders:
			sentence_count += 1

	calculate_readability(letter_count, word_count, sentence_count)

def calculate_readability(letter_count, word_count, sentence_count):
	L = (letter_count / word_count) * 100
	S = (sentence_count / word_count) * 100

	index = 0.0588 * L - 0.296 * S - 15.8

	if 1 <= index < 16:
		print(f"Grade {round(index)}")

	else:
		print("Before Grade 1") if index < 1 else print("Grade 16+")

main()
